src/main/java/com/amplitude/Event.java

```diff
@@ -9,56 +9,162 @@
 
 public class Event {
 
+  /** A unique identifier for your event. */
   public String eventType;
+  /**
+   * A readable ID specified by you. Must have a minimum length of 5 characters. Required unless
+   * device_id is present.
+   */
   public String userId;
+  /**
+   * A device-specific identifier, such as the Identifier for Vendor on iOS. Required unless user_id
+   * is present. If a device_id is not sent with the event, it will be set to a hashed version of
+   * the user_id.
+   */
   public String deviceId;
 
+  /**
+   * The timestamp of the event in milliseconds since epoch. If time is not sent with the event, it
+   * will be set to the request upload time.
+   */
   public long timestamp = System.currentTimeMillis();
+  /** The current Latitude of the user. */
   public double locationLat;
+  /** The current Longitude of the user. */
   public double locationLng;
 
+  /** The current version of your application. */
   public String appVersion;
+
   public String versionName;
 
+  /** Platform of the device. */
   public String platform;
+  /** The name of the mobile operating system or browser that the user is using. */
   public String osName;
+  /** The device brand that the user is using. */
   public String deviceBrand;
+  /** The device manufacturer that the user is using. */
   public String deviceManufacturer;
+  /** The device model that the user is using. */
   public String deviceModel;
+  /** The carrier that the user is using. */
   public String carrier;
 
+  /** The current country of the user. */
   public String country;
+  /** The current region of the user. */
   public String region;
+  /** The current city of the user. */
   public String city;
+  /** The current Designated Market Area of the user. */
   public String dma;
 
+  /** (iOS) Identifier for Advertiser. */
   public String idfa;
+  /** (iOS) Identifier for Vendor. */
   public String idfv;
+  /** (Android) Google Play Services advertising ID */
   public String adid;
+  /** (Android) Android ID (not the advertising ID) */
   public String androidId;
-
+  /** The (human) language set by the user. */
   public String language;
+  /**
+   * The IP address of the user. Use "$remote" to use the IP address on the upload request. We will
```

Comment:
What does `Use "$remote" to use the IP address on the upload request` mean?

Reply:
From HTTP v2 API. This means, pass "$remote" into this field to use the IP address of the HTTP request, as opposed to a custom set IP.